pkg/multicluster/kubeconfig/predicate.go

```diff
@@ -0,0 +1,36 @@
+package kubeconfig
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+var _ predicate.Predicate = SecretPredicate{}
+
+// SecretPredicate is a controller-runtime predicate that filters for KubeConfig secrets.
+type SecretPredicate struct{}
```

Comment:
does this need to be public?

Reply:
nope

Reply:
Instead you could use `predicate.Funcs` and just return an instance, rather than having a public struct